### PR TITLE
:book: Add a section for configuring Kubelet in Cluster API

### LINKED ISSUE #8425: Docs: Configuring Kubelet in Cluster API
### What would you like to be added (User Story)?

As a user I would like to understand how to configure Kubelet when deployed in a Cluster API cluster using KCP and CAPBK.

### Detailed Description

Currently there's a few different approaches:
1) Write a Kubelet conf using CloudInit files
2) Add Kubelet extraArgs in the KubeadmConfigTemplate
3) Patch kubeletconfiguration using a Kubeadm patch defined in the KubeadmConfigTemplate

The documentation should layout basic examples of each approach, and detail the limitations of each.

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind feature
/kind documentation

### PR DESCRIPTION
**Pull request by musaprg**
**What this PR does / why we need it**
This PR introduces a new page to describe how to configure kubelet with CAPBK.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8425

**TODOs**
- [x] Add examples
- [x] Add descriptions
- [x] Add limitations

**Considerations**

- [x] set [system-reserved](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) [[ref]](https://github.com/kubernetes-sigs/cluster-api/issues/8425#issuecomment-1531173401)

### CODE REVIEW

**Comment by musaprg** [comment]
I'm not confident about my English, so I would greatly appreciate your feedback on any grammatical errors.

File: docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+- [Pass `KubeletConfiguration` file via `KubeadmConfigSpec.files`](#pass-kubeletconfiguration-file-via-kubeadmconfigspecfiles)
+- [Set kubelet flags via `KubeadmConfigSpec.kubeletExtraArgs`](#set-kubelet-flags-via-kubeadmconfigspeckubeletextraargs)
+- [`kubeletconfiguration` patch target](#use-the-kubeletconfiguration-patch-target)
+
+## Pass `KubeletConfiguration` file via `KubeadmConfigSpec.files`
+
+You can use `KubeadmConfigSpec.files` to put any files on nodes. This example puts a `KubeletConfiguration` file on nodes via `KubeadmConfigSpec.files`, and makes kubelet use it via `KubeadmConfigSpec.kubeletExtraArgs`. You can check available configurations of `KubeletConfiguration` on [Kubelet Configuration (v1beta1) | Kubernetes](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration).
+
+This method is easy to replace the whole kubelet configuration, but it is not easy to replace only a part of the kubelet configuration.
```
Comment (fabriziopandini) [inline docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md:13]: Q: which kubelet configuration it is replacing? I assume the one generated by kubeadm, might be worth to make it explicit

File: docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+          # Set kubelet flags via KubeadmConfigSpec.kubeletExtraArgs
+          kubeletExtraArgs:
+            kube-reserved: cpu=1,memory=2Gi,ephemeral-storage=1Gi
+            system-reserved: cpu=500m,memory=1Gi,ephemeral-storage=1Gi
+            eviction-hard: memory.available<500Mi,nodefs.available<10%
+```
+
+## Use the `kubeletconfiguration` patch target
+
+We can use the `kubeletconfiguration` patch target to patch the kubelet configuration file. In this example, we put a patch file for `kubeletconfiguration` target in `strategic` `patchtype` on nodes via `KubeadmConfigSpec.files`. For more details, see [Customizing components with the kubeadm API | Kubernetes](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches)
```
Comment (fabriziopandini) [inline docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md:251]: nit
```suggestion
We can use kubeadm's `kubeletconfiguration` patch target to patch the kubelet configuration file. In this example, we put a patch file for `kubeletconfiguration` target in `strategic` `patchtype` on nodes via `KubeadmConfigSpec.files`. For more details, see [Customizing components with the kubeadm API | Kubernetes](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches)
```

File: docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        nodeRegistration:
+          criSocket: unix:///var/run/containerd/containerd.sock
+          # Set kubelet flags via KubeadmConfigSpec.kubeletExtraArgs
+          kubeletExtraArgs:
+            kube-reserved: cpu=1,memory=2Gi,ephemeral-storage=1Gi
+            system-reserved: cpu=500m,memory=1Gi,ephemeral-storage=1Gi
+            eviction-hard: memory.available<500Mi,nodefs.available<10%
+```
+
+## Use the `kubeletconfiguration` patch target
```
Comment (fabriziopandini) [inline docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md:249]:
```suggestion
## Use kubeadm's `kubeletconfiguration` patch target
```

File: docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+We can use the `kubeletconfiguration` patch target to patch the kubelet configuration file. In this example, we put a patch file for `kubeletconfiguration` target in `strategic` `patchtype` on nodes via `KubeadmConfigSpec.files`. For more details, see [Customizing components with the kubeadm API | Kubernetes](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches)
+
+This method is useful when you want to change the kubelet configuration file partially on specific nodes. For example, you can deploy a partially patched kubelet configuration file on specific nodes based on the default configuration used for `kubeadm init`.
```
Comment (fabriziopandini) [inline docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md:253]:
```suggestion
This method is useful when you want to change the kubelet configuration file partially on specific nodes. For example, you can deploy a partially patched kubelet configuration file on specific nodes based on the default configuration used for `kubeadm init` or `kubeadm join`.
```

**Review by fabriziopandini — COMMENTED** [review]
Thanks for writing up this doc!
Just a couple of nits from me

Looping in docs reviewers @oscr @elmiko

**Comment by snstaberah** [comment]
also need this func to config kubelet args

File: docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md (hunk 1 above)
Comment (musaprg) [reply docs/book/src/tasks/bootstrap/kubeadm-bootstrap/kubelet-config.md]: 3463620 fixed to make it clear where the default kubelet configuration comes from.

**Comment by musaprg** [comment]
> also need this func to config kubelet args

@snstaberah Which function are you referring to? It's possible that there was a permalink, but it appears to be missing.

**Comment by fabriziopandini** [comment]
lgtm pending squash
cc @sbueringer @killianmuldoon

**Review by killianmuldoon — COMMENTED** [review]
Great job - thanks for working on this @musaprg! Very clearly explained with good examples.

/lgtm
/approve

**Pull request merged**